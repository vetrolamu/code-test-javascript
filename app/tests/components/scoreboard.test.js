import React from 'react';
import Scoreboard  from '../../components/scoreboard/scoreboard.jsx';
import ScoreboardCellContent from '../../components/scoreboard/__cell/scoreboard__cell_type_content.jsx';
import renderer from 'react-test-renderer';

describe('Scoreboard', () => {
    it('ScoreboardCellContent renders correctly', () => {
        const cell1 = renderer.create(
            <ScoreboardCellContent active={false} />
        ).toJSON();
        expect(cell1).toMatchSnapshot();

        const cell2 = renderer.create(
            <ScoreboardCellContent active={true} rolls={[5, 5]} score={14} />
        ).toJSON();
        expect(cell2).toMatchSnapshot();

        const cell3 = renderer.create(
            <ScoreboardCellContent active={true} isLast={true} rolls={[5, 5, 3]} score={13} />
        ).toJSON();
        expect(cell3).toMatchSnapshot();
    });
    it('Scoreboard', () => {
        const scoreboardEmpty = renderer.create(
            <Scoreboard
                currentFrameIndex={0}
                currentPlayerIndex={0}
                playersNumber="1"
                results={[]}
            />
        ).toJSON();
        expect(scoreboardEmpty).toMatchSnapshot();
        const scoreboardFull = renderer.create(
            <Scoreboard
                currentFrameIndex={9}
                currentPlayerIndex={1}
                playersNumber="2"
                results={[[
                    {"rolls": [2, 3], "score": 5, "closed": true},
                    {"rolls": [10], "score": 24, "closed": true},
                    {"rolls": [10], "score": 20, "closed": true},
                    {"rolls": [4, 6], "score": 15, "closed": true},
                    {"rolls": [5, 2], "score": 7, "closed": true},
                    {"rolls": [4, 6], "score": 20, "closed": true},
                    {"rolls": [10], "score": 14, "closed": true},
                    {"rolls": [3, 1], "score": 4, "closed": true},
                    {"rolls": [5, 5], "score": 20, "closed": true},
                    {"rolls": [10, 5, 1], "score": 16, "closed": true}
                ],
				[
				    {"rolls": [4, 6], "score": 15, "closed": true},
				    {"rolls": [5, 5], "score": 20, "closed": true},
				    {"rolls": [10], "score": 20, "closed": true},
				    {"rolls": [3, 7], "score": 11, "closed": true},
				    {"rolls": [1, 9], "score": 13, "closed": true},
				    {"rolls": [3, 1], "score": 4, "closed": true},
				    {"rolls": [3, 7], "score": 20, "closed": true},
				    {"rolls": [10], "score": 16, "closed": true},
				    {"rolls": [5, 1], "score": 6, "closed": true},
				    {"rolls": [5], "score": 5, "closed": false}]
				]}
            />
        ).toJSON();
        expect(scoreboardFull).toMatchSnapshot();

    });
});
