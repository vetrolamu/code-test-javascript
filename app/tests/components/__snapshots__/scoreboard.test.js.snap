exports[`Scoreboard Scoreboard 1`] = `
<table
  className="scoreboard">
  <thead>
    <tr>
      <th
        className="scoreboard__cell">
        1
      </th>
      <th
        className="scoreboard__cell">
        2
      </th>
      <th
        className="scoreboard__cell">
        3
      </th>
      <th
        className="scoreboard__cell">
        4
      </th>
      <th
        className="scoreboard__cell">
        5
      </th>
      <th
        className="scoreboard__cell">
        6
      </th>
      <th
        className="scoreboard__cell">
        7
      </th>
      <th
        className="scoreboard__cell">
        8
      </th>
      <th
        className="scoreboard__cell">
        9
      </th>
      <th
        className="scoreboard__cell">
        10
      </th>
      <th
        className="scoreboard__cell" />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="scoreboard__cell scoreboard__cell_active scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_last scoreboard__cell_type_content">
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score" />
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_result">
        0
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Scoreboard Scoreboard 2`] = `
<table
  className="scoreboard">
  <thead>
    <tr>
      <th
        className="scoreboard__cell">
        1
      </th>
      <th
        className="scoreboard__cell">
        2
      </th>
      <th
        className="scoreboard__cell">
        3
      </th>
      <th
        className="scoreboard__cell">
        4
      </th>
      <th
        className="scoreboard__cell">
        5
      </th>
      <th
        className="scoreboard__cell">
        6
      </th>
      <th
        className="scoreboard__cell">
        7
      </th>
      <th
        className="scoreboard__cell">
        8
      </th>
      <th
        className="scoreboard__cell">
        9
      </th>
      <th
        className="scoreboard__cell">
        10
      </th>
      <th
        className="scoreboard__cell" />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          2
        </div>
        <div
          className="scoreboard__roll">
          3
        </div>
        <div
          className="scoreboard__frame-score">
          5
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          10
        </div>
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score">
          24
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          10
        </div>
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score">
          20
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          4
        </div>
        <div
          className="scoreboard__roll">
          6
        </div>
        <div
          className="scoreboard__frame-score">
          15
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__roll">
          2
        </div>
        <div
          className="scoreboard__frame-score">
          7
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          4
        </div>
        <div
          className="scoreboard__roll">
          6
        </div>
        <div
          className="scoreboard__frame-score">
          20
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          10
        </div>
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score">
          14
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          3
        </div>
        <div
          className="scoreboard__roll">
          1
        </div>
        <div
          className="scoreboard__frame-score">
          4
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__frame-score">
          20
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_last scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          10
        </div>
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__roll">
          1
        </div>
        <div
          className="scoreboard__frame-score">
          16
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_result">
        145
      </td>
    </tr>
    <tr>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          4
        </div>
        <div
          className="scoreboard__roll">
          6
        </div>
        <div
          className="scoreboard__frame-score">
          15
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__frame-score">
          20
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          10
        </div>
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score">
          20
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          3
        </div>
        <div
          className="scoreboard__roll">
          7
        </div>
        <div
          className="scoreboard__frame-score">
          11
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          1
        </div>
        <div
          className="scoreboard__roll">
          9
        </div>
        <div
          className="scoreboard__frame-score">
          13
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          3
        </div>
        <div
          className="scoreboard__roll">
          1
        </div>
        <div
          className="scoreboard__frame-score">
          4
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          3
        </div>
        <div
          className="scoreboard__roll">
          7
        </div>
        <div
          className="scoreboard__frame-score">
          20
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          10
        </div>
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score">
          16
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__roll">
          1
        </div>
        <div
          className="scoreboard__frame-score">
          6
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_active scoreboard__cell_last scoreboard__cell_type_content">
        <div
          className="scoreboard__roll">
          5
        </div>
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__roll" />
        <div
          className="scoreboard__frame-score">
          5
        </div>
      </td>
      <td
        className="scoreboard__cell scoreboard__cell_type_result">
        130
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Scoreboard ScoreboardCellContent renders correctly 1`] = `
<td
  className="scoreboard__cell scoreboard__cell_type_content">
  <div
    className="scoreboard__roll" />
  <div
    className="scoreboard__roll" />
  <div
    className="scoreboard__frame-score" />
</td>
`;

exports[`Scoreboard ScoreboardCellContent renders correctly 2`] = `
<td
  className="scoreboard__cell scoreboard__cell_active scoreboard__cell_type_content">
  <div
    className="scoreboard__roll">
    5
  </div>
  <div
    className="scoreboard__roll">
    5
  </div>
  <div
    className="scoreboard__frame-score">
    14
  </div>
</td>
`;

exports[`Scoreboard ScoreboardCellContent renders correctly 3`] = `
<td
  className="scoreboard__cell scoreboard__cell_active scoreboard__cell_last scoreboard__cell_type_content">
  <div
    className="scoreboard__roll">
    5
  </div>
  <div
    className="scoreboard__roll">
    5
  </div>
  <div
    className="scoreboard__roll">
    3
  </div>
  <div
    className="scoreboard__frame-score">
    13
  </div>
</td>
`;
